lib/spack/spack/test/cmd/test.py

```diff
@@ -319,3 +319,17 @@ def test_report_filename_for_cdash(install_mockery_mutable_config, mock_fetch):
     spack.cmd.common.arguments.sanitize_reporter_options(args)
     filename = spack.cmd.test.report_filename(args, suite)
     assert filename != "https://blahblah/submit.php?project=debugging"
+
+
+def test_test_output_multiple_specs(
+    mock_test_stage, mock_packages, mock_archive, mock_fetch, install_mockery_mutable_config
+):
+    """Ensure proper reporting for suite with skipped, failing, and passed tests."""
```

Comment:
Can you expand on "proper"? Also, it would be good to briefly explain the setup and the expected result in a comment.

Reply:
Great point.  Proper based on the idea that an enclosing test part's status should reflect that of the subparts in that sense that it needs to reflect failing subparts or when all subparts are skipped.  In other words, if there are any failed embedded test parts then the enclosing test part should be failed. And if all embedded test parts are skipped then the enclosing test part should be skipped.